tower-http/src/trace/body.rs

```diff
@@ -77,26 +77,30 @@ where
         self: Pin<&mut Self>,
         cx: &mut Context<'_>,
     ) -> Poll<Result<Option<HeaderMap>, Self::Error>> {
-        let this = self.project();
+        let mut this = self.project();
         let _guard = this.span.enter();
         let result = ready!(this.inner.poll_trailers(cx));
 
         let latency = this.start.elapsed();
 
-        if let Some((classify_eos, mut on_failure)) =
-            this.classify_eos.take().zip(this.on_failure.take())
```

Comment:
`this.on_failure.take()` meant we would only call `on_failure` for the first failure. That was a bug. It should be called for each failure.

Reply:
Looking at this again it still might not be correct since we're still doing `this.classify_eos.take()`. Meaning we can only classify one error and so the callback will still only be called once.

I've asked for clarification on Discord about what the contract of `http_body::Body` is.

Reply:
Hyper will not call `poll_data` again if it returns an error. So this is fine.